test/benchmarks/netperf/launch.go

```diff
@@ -0,0 +1,430 @@
+}
+
+func setupClient() *client.Client {
+	// Read user's kube config and use the server and credentials from the current context
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	kubeConfig, err := loadingRules.Load()
+	if err != nil {
+		fmt.Println("Failure loading kubeconfig credentials and server location")
+		return nil
+	}
+
+	context, ok := kubeConfig.Contexts[kubeConfig.CurrentContext]
+	if !ok {
+		fmt.Println("Failed to find default kubectl context")
+		return nil
+	}
+	authInfo, ok := kubeConfig.AuthInfos[fmt.Sprintf("%s-basic-auth", context.AuthInfo)]
+	if !ok {
+		fmt.Println("Failed to find the default auth info in the kubectl config file")
+		return nil
+	}
+
+	fmt.Printf("Setting up kubeConfig parameters for REST client User %s Password ***** Host: %s\n",
+		authInfo.Username,
+		kubeConfig.Clusters[kubeConfig.Contexts[kubeConfig.CurrentContext].Cluster].Server)
+
+	config := &restclient.Config{
+		Username: authInfo.Username,
+		Password: authInfo.Password,
+		Host:     kubeConfig.Clusters[kubeConfig.Contexts[kubeConfig.CurrentContext].Cluster].Server,
+		Insecure: true,
+	}
+	c, err := client.New(config)
+	if err != nil {
+		fmt.Println("Failed to connect to Kubernetes API server", err)
+		return nil
+	}
+	return c
+}
+
+// getMinions : Only return minion nodes
+func getMinionNodes(c *client.Client) *api.NodeList {
+	var nodesTmp *api.NodeList
+	var nodes api.NodeList
+	var err error
+	if nodesTmp, err = c.Nodes().List(everythingSelector); err != nil {
+		fmt.Println("Failed to fetch nodes", err)
+		return nil
+	}
+
+	for _, n := range nodesTmp.Items {
+		if strings.Contains(n.GetName(), "-minion") {
+			nodes.Items = append(nodes.Items, n)
+		}
+	}
+
+	return &nodes
+}
+
+func cleanup(c *client.Client) {
+	// Cleanup existing rcs, pods and services in our namespace
+	rcs, err := c.ReplicationControllers(testNamespace).List(everythingSelector)
+	if err != nil {
+		fmt.Println("Failed to get replication controllers", err)
+		return
+	}
+	for _, rc := range rcs.Items {
+		fmt.Println("Deleting rc", rc.GetName())
+		if err := c.ReplicationControllers(testNamespace).Delete(rc.GetName()); err != nil {
+			fmt.Println("Failed to delete rc", rc.GetName(), err)
+		}
+	}
+	pods, err := c.Pods(testNamespace).List(everythingSelector)
+	if err != nil {
+		fmt.Println("Failed to get pods", err)
+		return
+	}
+	for _, pod := range pods.Items {
+		fmt.Println("Deleting pod", pod.GetName())
+		var gracePeriodZero int64
+		if err := c.Pods(testNamespace).Delete(pod.GetName(), &api.DeleteOptions{GracePeriodSeconds: &gracePeriodZero}); err != nil {
+			fmt.Println("Failed to delete pod", pod.GetName(), err)
+		}
+	}
+	svcs, err := c.Services(testNamespace).List(everythingSelector)
+	if err != nil {
+		fmt.Println("Failed to get services", err)
+		return
+	}
+	for _, svc := range svcs.Items {
+		fmt.Println("Deleting svc", svc.GetName())
+		c.Services(testNamespace).Delete(svc.GetName())
+	}
+	time.Sleep(10 * time.Second)
```

Comment:
why the sleep? there's not guarantee pods you find will have 10s grace period, and if they're pods you created you can simply create them with gracePeriod=0?